### PR TITLE
feat: ability for fedimint-cli to call client modules

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: client/cli/src/lib.rs [hunk 1]
```diff
@@ -242,29 +245,51 @@ impl Opts {
...
+        module_gens: &ClientModuleGenRegistry,
+    ) -> ModuleDecoderRegistry {
+        ModuleDecoderRegistry::new(cfg.clone().0.modules.into_iter().filter_map(
             |(id, module_cfg)| {
-                let module_gen = module_gens
+                module_gens
                     .get(module_cfg.kind())
-                    .expect("module kind not found in registry");
-                (id, module_gen.as_ref().decoder())
+                    .map(|module_gen| (id, module_gen.as_ref().decoder()))
```
Comment (justinmoon) [inline client/cli/src/lib.rs:276]: Nice, this makes it so unknown modules just get ignored instead of panicing

File: client/cli/src/lib.rs [hunk 2]
```diff
@@ -953,6 +1025,37 @@ impl FedimintCli {
                     .await?;
                 Ok(CliOutput::EpochCount { count })
             }
+            Command::Module { id, arg } => {
+                let cfg = cli.load_config_or_terminate();
+                let decoders = cli.load_decoders(&cfg, &self.module_gens);
+                let db = cli.load_db_or_terminate(&decoders);
+                let (_id, module_cfg) = match id {
+                    ModuleSelector::Id(id) => (
+                        id,
```
Comment (justinmoon) [inline client/cli/src/lib.rs:1034]: If this isn't used, maybe don't return it?

**Review by justinmoon — APPROVED** [review]
one small nit

File: client/cli/src/lib.rs (hunk 2 above)
Comment (dpc) [reply client/cli/src/lib.rs:1034]: Then I have to do an extra `.map` to get rid of it on the other case, so this is actually shorter. :D

File: client/cli/src/lib.rs [hunk 3]
```diff
@@ -953,6 +1025,37 @@ impl FedimintCli {
...
+                        cfg.as_ref()
+                            .get_module_cfg(id)
+                            .or_terminate_msg(CliErrorKind::IOError, "Can't load module"),
+                    ),
+                    ModuleSelector::Kind(kind) => cfg
+                        .as_ref()
+                        .get_first_module_by_kind_cfg(kind)
```
Comment (jkitman) [inline client/cli/src/lib.rs:1041]: So is getting by kind just to make the command more user-friendly?

File: modules/fedimint-mint-client/src/lib.rs [hunk 4]
```diff
@@ -23,7 +23,7 @@ impl ClientModuleGen for MintClientGen {
     type Config = MintClientConfig;
 
     async fn init(&self, _cfg: Self::Config, _db: Database) -> anyhow::Result<Self::Module> {
-        unimplemented!()
+        Ok(MintClientModule {})
```
Comment (jkitman) [inline modules/fedimint-mint-client/src/lib.rs:26]: Why only implement for the mint module?

File: client/cli/src/lib.rs (hunk 3 above)
Comment (dpc) [reply client/cli/src/lib.rs:1041]: Yeap. I suspect in majority of cases people will have one module instance of a given kind, so referring to it by kind does seem to make things smoother ... .

We could possibly change it to be "by kind, only if it is not ambiguous" or something.

**Pull request merged**